subsys/usb/device_next/class/usbd_cdc_ecm.c

```diff
@@ -504,6 +506,7 @@ static int cdc_ecm_send(const struct device *dev, struct net_pkt *const pkt)
 		return -EACCES;
 	}
 
+	ep = cdc_ecm_get_bulk_out(c_data);
 	buf = cdc_ecm_buf_alloc(cdc_ecm_get_bulk_in(c_data));
```

Comment:
It should be IN endpoint, like in the line below.

```suggestion
	ep = cdc_ecm_get_bulk_in(c_data);
	buf = cdc_ecm_buf_alloc(ep);
```